src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -1138,6 +1138,39 @@ public TimeSpan ErrorEventsHarvestCycle
         public virtual uint ErrorsMaximumPerPeriod { get { return 20; } }
         public virtual IEnumerable<string> ExceptionsToIgnore { get { return ServerOverrides(_serverConfiguration.RpmConfig.ErrorCollectorErrorsToIgnore, _localConfiguration.errorCollector.ignoreErrors.exception); } }
 
+        public virtual IEnumerable<string> ExpectedClasses { get { return ServerOverrides(_serverConfiguration.RpmConfig.ErrorCollectorExpectedClasses, _localConfiguration.errorCollector.expectedClasses.errorClass); } }
+
+        public virtual IEnumerable<KeyValuePair<string, IEnumerable<string>>> ExpectedMessages
+        {
+            get
+            {
+                var expectedMessages = new List<KeyValuePair<string, IEnumerable<string>>>();
+
+                foreach(var errorClass in _localConfiguration.errorCollector.expectedMessages)
+                {
+                    var messages = errorClass.message;
+                    if (messages != null)
+                    {
+                        expectedMessages.Add(new KeyValuePair<string, IEnumerable<string>>(errorClass.name, messages));
+                    }
+                }
+                return ServerOverrides(_serverConfiguration.RpmConfig.ErrorCollectorExpectedMessages, expectedMessages);
```

Comment:
We should probably cache the results of this computation so that we don't re-evaluate this logic with every error on every transaction.